### PR TITLE
Add default template to search application

### PR DESCRIPTION
**Pull request by kderusso**
Adds a simple default search template to search applications if none are specified, and adds tests for search applications created with a string mustache script input. 

If you create/update a search application that does not specify or removes an existing template from the search application, we will create a default template: 
```
"template": {
      "script": {
        "source": """
        {
          "query": {
            "query_string":{
              "query": "{{query_string}}",
              "default_field": "{{default_field}}"
            }
          }
        }
        """,
        "params": {
          "query_string": "*",
          "default_field": "*"
        }
      }
    }
```

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplate.java [hunk 1]
```diff
@@ -96,4 +99,12 @@ public boolean equals(Object o) {
     public Script script() {
         return script;
     }
+
+    private static SearchApplicationTemplate buildDefaultTemplate() {
+        Map<String, Object> params = Map.of("query_string", "*", "default_field", "*");
+        String query = "{ \"query\": { \"query_string\":{ \"query\": \"{{query_string}}\", \"default_field\": \"{{default_field}}\" } } }";
```
Comment (jimczi) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplate.java:105]: nit: Can you use the `"""` notation and avoid the escape `"` path?

Comment (jimczi) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplate.java:105]: We can also keep the json indentation to make it clearer

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplate.java [hunk 2]
```diff
@@ -30,6 +31,8 @@
  * {@link QuerySearchApplicationAction}, overriding the parameters included on it via {@link Request}
  */
 public class SearchApplicationTemplate implements ToXContentObject, Writeable {
+
+    public static final SearchApplicationTemplate DEFAULT_TEMPLATE = buildDefaultTemplate();
```
Comment (jimczi) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplate.java:35]: nit: we probably don't need a function and can set the definition inline with the final variable?

**Review by jimczi — APPROVED** [review]
LGTM, added some code cosmetic comments

**Review by carlosdelest — APPROVED** [review]

**Review by ioanatia — APPROVED** [review]

**Pull request merged**